flamingpy/utils/viz.py

```diff
@@ -284,6 +329,109 @@ def draw_EGraph(
     return fig, ax
 
 
+def draw_EGraph_plotly(
+    egraph,
+    color_nodes=False,
+    color_edges=False,
+    label="coordinates",
+    title=None,
+    legend=False,
+    show_axes=True,
+    **kwargs,
+):
+    """Draw the graph state represented by the EGraph with plotly.
+
+    Args:
+        See draw_EGraph for general keyword arguments, see keyword arguments below for
+            matplotlib-specific arguments.
+
+    Keyword args:
+        showbackground (bool): if True, shows the background of the graph. Default is False.
+        showspikes (bool): if True, shows coordinate-lines when hovering over a node. Default is
+            False.
+        showgrid (bool): if True, shows the grid of the graph. Default is True.
+        width (int): width of the graph. Default is 750.
+        height (int): height of the graph. Default is 750.
+
+    Returns:
+        figure: Plotly Express figure.
+    """
+    # nodes
+    nodes = np.array(egraph.nodes)
+    x_nodes, y_nodes, z_nodes = nodes[:, 0], nodes[:, 1], nodes[:, 2]
+
+    nodeColors = [_get_node_color(egraph, node, color_nodes) for node in egraph.nodes]
+    nodeInfo = [_get_node_info(egraph, node, information=label) for node in egraph.nodes]
+
+    node_trace = go.Scatter3d(
+        name="nodes",
+        x=x_nodes,
+        y=y_nodes,
+        z=z_nodes,
+        mode="markers",
+        marker=dict(
+            symbol="circle",
+            size=5,
+            color=nodeColors,
+            line=dict(color="black", width=0.5),
+        ),
+        hovertext=nodeInfo,
+        hoverinfo="text",
+    )
+
+    # edges
+    x_edges, y_edges, z_edges = [], [], []
+
+    for edge in egraph.edges:
+        x0, y0, z0 = edge[0]
+        x1, y1, z1 = edge[1]
+        x_edges.extend([x0, x1, None])
+        y_edges.extend([y0, y1, None])
+        z_edges.extend([z0, z1, None])
+
+    edgeColors = [_get_edge_color(egraph, edge, color_edges) for edge in egraph.edges]
+
+    edge_trace = go.Scatter3d(
+        name="edges",
+        x=x_edges,
+        y=y_edges,
+        z=z_edges,
+        line=dict(color="black", width=1),
+        mode="lines",
+        marker=dict(color=edgeColors),
+        hoverinfo="none",
+    )
+
+    # axis
+    axis = dict(
+        showbackground=kwargs.get("showbackground", False),
+        showline=True,
+        showspikes=kwargs.get("showspikes", False),
+        zeroline=False,
+        showgrid=kwargs.get("showgrid", True),
+        showticklabels=show_axes,
+        tickmode="linear",
+        tick0=0,
+        dtick=1,
+    )
+
+    # layout
+    layout = go.Layout(
+        width=kwargs.get("width", 750),
+        height=kwargs.get("height", 750),
+        showlegend=legend,
+        hovermode="closest",
+        scene=dict(
+            xaxis={**dict(title="x"), **axis},
+            yaxis={**dict(title="y"), **axis},
+            zaxis={**dict(title="z"), **axis},
+        ),
+        title=_get_title(title, label),
```

Comment:
In principle, `label` could be a list, right? In which case `_get_title `would return the list (since it is not in the default dictionary) and plotly would complain that it's getting a list.

Reply:
If `label` is a list, the user should provide an explicit title themselves. The `_get_title` function only facilitates easy titling in cases only one attribute is shown. If the user tries to set `label` to a list/tuple and set `title=True`, the resulting title would be `None` as mentioned in the docstring.

Reply:
alternatively, I could set `title=label`, but I don't think that's visually appealing. What do you think @ilan-tz?

Reply:
I now unpack `label` if it's a list/tuple and show it like `"a, b, c"` when `label = ["a", "b", "c"]`